src/routes/forms/submission/submissionName/confirm/router.ts

```diff
@@ -1,8 +1,52 @@
-import { Router } from "express";
-import { confirmationCodeApiRoute } from "@src/routes/forms/submission/submissionName/confirm/confirmationCode/router";
+import { confirmFormSubmission } from "@src/lib/vault/confirmFormSubmission";
+import {
+  FormSubmissionAlreadyConfirmedException,
+  FormSubmissionNotFoundException,
+  FormSubmissionIncorrectConfirmationCodeException,
+} from "@src/lib/vault/dataStructures/exceptions";
+import { type Request, type Response, Router } from "express";
 
 export const confirmApiRoute = Router({
   mergeParams: true,
 });
 
-confirmApiRoute.use("/:confirmationCode", confirmationCodeApiRoute);
+confirmApiRoute.put(
+  "/:confirmationCode([a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12})",
+  async (request: Request, response: Response) => {
+    const formId = request.params.formId;
+    const submissionName = request.params.submissionName;
+    const confirmationCode = request.params.confirmationCode;
+
+    try {
+      await confirmFormSubmission(formId, submissionName, confirmationCode);
+      return response.sendStatus(200);
+    } catch (error) {
+      if (error instanceof FormSubmissionNotFoundException) {
+        return response
+          .status(404)
+          .json({ error: "Form submission does not exist" });
+      }
+
+      if (error instanceof FormSubmissionAlreadyConfirmedException) {
+        return response
+          .status(200)
+          .json({ info: "Form submission is already confirmed" });
```

Comment:
Decided to return a `200` response with an associated info message for when someone tries to confirm an already confirmed form submission.
Open to discussion if we think it would be better to let the user know what actually happened at the API/database level with an error response.

Reply:
just thinking if as a client Id like to track successful confirm in an execution, I would now need to parse the json. I can't find a meaningful http code to replace it

Reply:
We could see the JSON message as an optional piece of information if someone really needs to track when a confirmation is done multiple times on the same item but otherwise you can just ignore it as it will be a `200` response code anyway.

Reply:
The `200` makes sense to me - we could always switch this in the future to a `400` if it's something we start seeing a lot of.